framework/helpers/BaseHtml.php

```diff
@@ -884,9 +884,12 @@ public static function checkboxList($name, $selection = null, $items = [], $opti
             $name .= '[]';
         }
 
-        $formatter = isset($options['item']) ? $options['item'] : null;
-        $itemOptions = isset($options['itemOptions']) ? $options['itemOptions'] : [];
-        $encode = !isset($options['encode']) || $options['encode'];
+        $formatter = ArrayHelper::remove($options, 'item');
+        $itemOptions = ArrayHelper::remove($options, 'itemOptions', []);
+        $encode = ArrayHelper::remove($options, 'encode', true);
+        $separator = ArrayHelper::remove($options, 'separator', PHP_EOL);
```

Comment:
that is describing the opposite of what want to imply. the solution described there is to use "\r\n" instead of PHP_EOL or "\n"

The issue is that a file is being generated on linux and then its downloaded and got open on windows. the line separator on that file is "\n" so windows see it as a continuos line.

so the solution was to use "\r\n" instead of PHP_EOL or "\n" (which causes the same problem).

thats the only case in which it would happen and if you wanna fix it we should use "\r\n" by default instead of the other two options.

This change WILL make the tests fail for sure.